lib/db_connection/app.ex

```diff
@@ -6,10 +6,11 @@ defmodule DBConnection.App do
 
   def start(_, _) do
     children = [
+      worker(DBConnection.Watcher, []),
       supervisor(DBConnection.Task, []),
       supervisor(DBConnection.Sojourn.Supervisor, []),
       supervisor(DBConnection.Ownership.PoolSupervisor, [])
-   ]
-    Supervisor.start_link(children, [strategy: :one_for_one, name: __MODULE__])
+    ]
+    Supervisor.start_link(children, strategy: :rest_for_one, name: __MODULE__)
```

Comment:
This supervision tree isn't quite right because the `Watcher` makes calls to the other supervisors. Therefore it depends on those supervisors and so we want the `Watcher` to be started last/shutdown first. Can we put it last and make this a `:one_for_all`. Therefore we would still shutdown all pools if the `Watcher` crashes. None of the supervisors should ever crash anyway because they all have temporary children.